pages/[user]/[type].tsx

```diff
@@ -37,6 +39,26 @@ export default function Type(props): Type {
     setSelectedDate(date);
   };
 
+  useEffect(() => {
+    if (!selectedDate) {
+      return;
+    }
+
+    const formattedDate = selectedDate.utc().format("YYYY-MM-DD");
+
+    router.replace(
+      {
+        query: {
+          date: formattedDate,
+        },
+      },
+      undefined,
+      {
+        shallow: true,
+      }
+    );
+  }, [selectedDate]);
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'router'. Either include it or remove the dependency array.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7748664)
```suggestion
  }, [router, selectedDate]);
```